RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1481 @@
+        Vec3D const threads_3d{channelsPerBlock, mahiPulseOffsets.metadata().size(), windowSize};
+
+        auto workDivPrep3D = cms::alpakatools::make_workdiv<Acc3D>(blocks_3d, threads_3d);
+
+        //Device buffer for output
+        auto pulseMatrices = cms::alpakatools::make_device_buffer<float[]>(
+            queue, totalChannels * windowSize * mahiPulseOffsets.metadata().size());
+        auto pulseMatricesM = cms::alpakatools::make_device_buffer<float[]>(
+            queue, totalChannels * windowSize * mahiPulseOffsets.metadata().size());
+        auto pulseMatricesP = cms::alpakatools::make_device_buffer<float[]>(
+            queue, totalChannels * windowSize * mahiPulseOffsets.metadata().size());
+
+        alpaka::exec<Acc3D>(queue,
+                            workDivPrep3D,
+                            mahi::kernel_prep_pulseMatrices_sameNumberOfSamples{},
+                            reinterpret_cast<float*>(pulseMatrices.data()),
+                            reinterpret_cast<float*>(pulseMatricesM.data()),
+                            reinterpret_cast<float*>(pulseMatricesP.data()),
+                            mahiPulseOffsets,
+                            reinterpret_cast<float*>(amplitudes.data()),
+                            f01HEDigis,
+                            f5HBDigis,
+                            f3HBDigis,
+                            reinterpret_cast<float*>(soiSamples.data()),
+                            mahi,
+                            recoParamsWithPS,
+                            configParameters.meanTime,
+                            configParameters.timeSigmaSiPM,
+                            configParameters.timeSigmaHPD,
+                            configParameters.applyTimeSlew,
+                            configParameters.tzeroTimeSlew,
+                            configParameters.slopeTimeSlew,
+                            configParameters.tmaxTimeSlew);
+
+        uint32_t threadsPerBlock = configParameters.kernelMinimizeThreads[0];
+        auto blocks_1d = cms::alpakatools::divide_up_by(totalChannels, threadsPerBlock);
+
+        auto workDivPrep1D = cms::alpakatools::make_workdiv<Acc1D>(blocks_1d, threadsPerBlock);
+
+        alpaka::exec<Acc1D>(queue,
+                            workDivPrep1D,
+                            mahi::kernel_minimize<8, 8>{},
+                            outputGPU,
+                            reinterpret_cast<float*>(amplitudes.data()),
+                            reinterpret_cast<float*>(pulseMatrices.data()),
+                            reinterpret_cast<float*>(pulseMatricesM.data()),
+                            reinterpret_cast<float*>(pulseMatricesP.data()),
+                            mahiPulseOffsets,
+                            reinterpret_cast<float*>(noiseTerms.data()),
+                            reinterpret_cast<float*>(electronicNoiseTerms.data()),
+                            reinterpret_cast<float*>(soiSamples.data()),
```

Comment:
Same set of comments on the `reinterpret_cast`s.

Reply:
removed